src/gui/DisplayPanel.cpp

```diff
@@ -904,6 +904,7 @@ void DisplayPanel::OpenFile(wxString wanted_filename, wxString wanted_tab_title,
     else {
         my_notebook->Freeze( );
         my_notebook->AddPage(my_panel, wanted_tab_title, false);
+        my_notebook->SetPageToolTip(panel_counter - 1, my_panel->filename);
```

Comment:
My only comment is that panel_counter - 1 depends on the same else{} condition incrementing panel_counter in line 839-855

This seems like a place to introduce a bug in the future if either of the *if* conditions are changed.

consider setting the first one on to something like

bool useful_name = (condition)

then checking if (useful_name) in the first block, and if (! useful_name) here, and getting rid of the empty true block 

That way, if (condition) is ever changed, it will self propagate

Reply:
This makes sense; I wonder if in general it would be better to simply add a bool for all possible flags as a member of the DisplayPanel class, and perform the flag checks once in `DisplayPanel::Initialise`, assigning each to a bool that can subsequently be used where needed -- or at least for those flags that are also checked in other functions or multiple times?

Reply:
I guess the real question is does the whole system make any sense? My opinion (and maybe it isn't Tim G's) is that this use of bit-wise flags is a cute parlor trick that is more confusing than it is efficient. I.e., IMO, I don't think there is any benefit to setting all the flags in one variable's bits (style_flags) and then using the fairly confusing

(style_flags & SOME_INT) == SOME_INT 

to check if that particular bit is set.

I find it hard to believe this is measurably more efficient than just taking all those defines we're storing in bits

```cpp
#define CAN_CHANGE_FILE 1 // 2^0, bit 0
#define CAN_CLOSE_TABS 2 // 2^1, bit 1
#define CAN_FFT 4 // 2^2, bit 2
#define START_WITH_INVERTED_CONTRAST 8
#define START_WITH_AUTO_CONTRAST 16
#define NO_NOTEBOOK 32
#define FIRST_LOCATION_ONLY 64
#define START_WITH_FOURIER_SCALING 128
#define DO_NOT_SHOW_STATUS_BAR 256
#define CAN_SELECT_IMAGES 1024
#define NO_POPUP 2048
#define START_WITH_NO_LABEL 4096
#define SKIP_LEFTCLICK_TO_PARENT 8192
#define CAN_MOVE_TABS 16384
#define DRAW_IMAGE_SEPARATOR 32768
#define KEEP_TABS_LINKED_IF_POSSIBLE 65536
```

and instead define them as (private) boolean members of the DisplayPanel class.

Then, when calling Initialse, just set them like a sane person:

```cpp

this->cisTEMDisplayPanel->Initialise(CAN_CHANGE_FILE | CAN_CLOSE_TABS | CAN_MOVE_TABS | CAN_FFT);

// becomes

this->cisTEMDisplayPanel->Initialise()
this->cisTEMDisplayPanel->can_change_file = True;
this->cisTEMDisplayPanel->can_close_tabs = True;
this->cisTEMDisplayPanel->can_move_tabs = True;
this->cisTEMDisplayPanel->can_fft = True;

```

Reply:
*I guess you would need setters if they were private (or make DisplayFrame a friend of DisplayPanel)

For simplicity, I guess public members would be okay too